### ISSUE TITLE
Bug in function ZIP

### ISSUE THREAD

**Issue by miguel-porto** [arangodb/arangodb#1722]
Hi there,

I think I found a bug here. If I do
```
RETURN ZIP(['a11111','b222','c'],['value','value','value'])
```
It works as expected. If instead I do the equivalent:
```
LET val=(FOR i in 1..3 RETURN 'value')
RETURN ZIP(['a11111','b222','c'],val)
```
the output is
```
[
  {
    "a11111": "value",
    "b22211": "value",
    "c22211": "value"
  }
]
```

It seems the string buffer is not cleared!
Best,
Miguel

**Comment by jsteemann** [comment]
Thanks. Fixed in 2.8 and devel branches.

**Comment by miguel-porto** [comment]
Thanks!!
